src/config/machines/example-grant-machine.js

```diff
@@ -171,9 +174,29 @@ export const exampleGrantMachine = createMachine({
         ...pageUIConfig.consent
       }
     },
-
     confirmation: {
-      type: 'final',
+      initial: 'submitting',
+      states: {
+        submitting: {
+          invoke: {
+            // eslint-disable-next-line jsdoc/require-jsdoc
+            src: (context) => {
+              if (!context.hasSubmitted) {
+                handleSubmission(context);
```

Comment:
Could we move this into its dedicated `services` section of the state machine definition?

```
services: {
    handleSubmission: (context) => {
      if (!context.hasSubmitted) {
        return handleSubmission(context).then(() => {
          context.hasSubmitted = true;
        });
      }
    },
  },
```

and then just:

```
confirmation: {
      initial: 'submitting',
      states: {
        submitting: {
          invoke: {
            src: 'handleSubmission',
            onDone: {
              target: 'success',
            },
            onError: {
              actions: (_, event) =>
                console.error('Submission failed:', event.data),
            },
          },
        },
        success: {
          type: 'final',
        },
      },
      meta: {
        templateId: 'confirmation',
        currentPageId: 'confirmation',
        ...pageUIConfig.confirmation,
      },
    },
```